tests/inputfiles/compact_test3.txt

```diff
@@ -0,0 +1 @@
+85283473fffffff85283447fffffff8528347bfffffff85283463fffffff85283477fffffff8528340ffffffff8528340bfffffff85283457fffffff85283443fffffff8528344ffffffff852836b7fffffff8528346bfffffff8528346ffffffff85283467fffffff8528342bfffffff8528343bfffffff85283407fffffff85283403fffffff8528341bfffffff
```

Comment:
Do we really want to support this kind of fixed length record format? This seems like the worst of human readability from binary files (hard to know if the records are the right size, as opposed to e.g. JSON or CSV where you can eyeball and see where the record separators are) and parsing performance of text files. (can't just directly reference this as a buffer of `(u)int64`, must parse to integers)

Reply:
I simplified it a bit (and moved it to a singular function), but I couldn't reduce the complexity as much as I wanted to because `sscanf` returns the *number* of matched variables, not the number of matched characters in the string, so I can't advance the string in the obvious way.

I am half-tempted to rewrite it without any `sscanf` call and build the integer manually.